src/spaces/compartment.jl

```diff
@@ -0,0 +1,188 @@
+  pos = agent.pos .+ dt .* agent.vel
+  if model.space.periodic
+    pos = mod.(pos, model.space.dims)
+  end
+  move_agent!(agent, pos, model)
+  return agent.pos
+end
+
+"""
+    move_agent!(agent::A, model::ABM{A, CompartmentSpace}, vel::NTuple{D, N}, dt = 1.0)
+Propagate the agent forwards one step according to `vel` and the model's space, with `dt` as the time step. (`update_vel!` is not used)
+"""
+function move_agent!(agent::A, model::ABM{A,S,F,P}, vel::NTuple{D, X}, dt = 1.0) where {A <: AbstractAgent, S <: CompartmentSpace, F, P, D, X <: AbstractFloat}
+  pos = agent.pos .+ dt .* vel
+  if model.space.periodic
+    pos = mod.(pos, model.space.dims)
+  end
+  move_agent!(agent, pos, model)
+  return agent.pos
+end
+
+#######################################################################################
+# %% Neighbors and stuff
+#######################################################################################
+
+grid_space_neighborhood(α, model::ABM{<:AbstractAgent, <:CompartmentSpace}, r) =
+  grid_space_neighborhood(α, model.space.grid, r)
+
+function nearby_ids(pos::ValidPos, model::ABM{<:AbstractAgent,<:CompartmentSpace}, r = 1)
+  nn = grid_space_neighborhood(CartesianIndex(pos), model, r)
+  s = model.space.grid.s
+  Iterators.flatten((s[i...] for i in nn))
+end
+
+function nearby_positions(pos::ValidPos, model::ABM{<:AbstractAgent,<:CompartmentSpace}, r = 1)
+  nn = grid_space_neighborhood(CartesianIndex(pos), model, r)
+  Iterators.filter(!isequal(pos), nn)
+end
+
+function positions(model::ABM{<:AbstractAgent,<:CompartmentSpace})
+  x = CartesianIndices(model.space.grid.s)
+  return (Tuple(y) for y in x)
+end
+
+function ids_in_position(pos::ValidPos, model::ABM{<:AbstractAgent,<:CompartmentSpace})
+  return model.space.grid.s[pos...]
+end
+
+cell_center(pos::Tuple) = getindex.(modf.(pos), 2) .+ 0.5
+distance_from_cell_center(pos::Tuple, center) = sqrt(sum(abs2.(pos .- center)))
+
+"""
+    space_neighbors(position, model::ABM, r=1; kwargs...) → ids
+
+Return an iterator of the ids of the agents within "radius" `r` of the given `position`
+(which must match type with the spatial structure of the `model`).
+
+What the "radius" means depends on the space type:
+- `GraphSpace`: `r` means the degree of neighbors in the graph and is an integer.
+  For example, for `r=2` include first and second degree neighbors.
+- `GridSpace, ContinuousSpace`: Standard distance implementation according to the
+  underlying space metric.
+
+## Keywords
+Keyword arguments are space-specific.
+For `GraphSpace` the keyword `neighbor_type=:default` can be used to select differing
+neighbors depending on the underlying graph directionality type.
+- `:default` returns neighbors of a vertex. If graph is directed, this is equivalent
+  to `:out`. For undirected graphs, all options are equivalent to `:out`.
+- `:all` returns both `:in` and `:out` neighbors.
+- `:in` returns incoming vertex neighbors.
+- `:out` returns outgoing vertex neighbors.
+"""
```

Comment:
This documentation string should not be duplicated here, right?

Reply:
No, I will fix the doc strings.